### PR TITLE
[snapshots] Address pending TODOs

### PR DESCRIPTION
**Pull request by ricab**

### CODE REVIEW

File: src/platform/backends/shared/base_virtual_machine.cpp [hunk 1]
```diff
@@ -526,8 +526,6 @@ void BaseVirtualMachine::restore_snapshot(const std::string& name, VMSpecs& spec
     assert_vm_stopped(state); // precondition
 
     auto snapshot = get_snapshot(name);
-
-    // TODO@snapshots convert into runtime_error (persisted info could have been tampered with)
```
Comment (sharder996) [inline src/platform/backends/shared/base_virtual_machine.cpp:530]: What was the reason/resolution for removing this?

**Review by sharder996 — COMMENTED** [review]
Looks good! Just a few comments

File: src/daemon/instance_settings_handler.h [hunk 2]
```diff
@@ -18,11 +18,11 @@
 #ifndef MULTIPASS_INSTANCE_SETTINGS_HANDLER_H
 #define MULTIPASS_INSTANCE_SETTINGS_HANDLER_H
 
-#include "vm_specs.h"
```
[the remark this answers is not in the document]
Comment (ricab) [reply src/daemon/instance_settings_handler.h:21]: Yup sure.

File: src/platform/backends/shared/base_virtual_machine.cpp (hunk 1 above)
Comment (ricab) [reply src/platform/backends/shared/base_virtual_machine.cpp:530]: I implemented it in the snapshot: https://github.com/canonical/multipass/pull/3265/commits/d007d26db7098f8ebc086af1ada275b0f34532f9

Better fail early in this case and refuse to load the snapshot at all than attempt to run with faulty contents.

Comment (sharder996) [reply src/platform/backends/shared/base_virtual_machine.cpp:526]: Got it! Thanks for the explanation

**Review by sharder996 — APPROVED** [review]

**Pull request merged**